File: Solutions/1768/1768.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1768.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-10-23 17:39:46
***************************************************************************************************/


char * mergeAlternately(char * word1, char * word2){
    int n1 = strlen(word1);
    int n2 = strlen(word2);
    char * ret = (char *)malloc(sizeof(char) * (n1+n2+1));
    // char * ret = (char *)malloc(sizeof(char) * (strlen(word1) + strlen(word2) + 1));
    int i = 0;
    int j = 0;
    while (j < strlen(word1) || j < strlen(word2)) {
        if (j < strlen(word1)) {
            ret[i++] = word1[j];
        }
        if (j < strlen(word2)) {
            ret[i++] = word2[j];
        }
        j++;
    }
    ret[i] = '\0';
    return ret;
}

File: Solutions/1768/1768.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/10/23 10:03
# File: 1768.py
# Desc: 

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        s = list()
        n = min(len(word1), len(word2))
        for i in range(n):
            s.append(word1[i])
            s.append(word2[i])
        return ''.join(s) + word1[n:] + word2[n:]
